mlflow/pyspark/ml/__init__.py

```diff
@@ -1039,24 +1059,16 @@ def _log_posttraining_metadata(estimator, spark_model, params, input_df):
                     cast_spark_df_with_vector_to_array,
                     get_feature_cols,
                 )
-                from mlflow.spark import _find_and_set_features_col_as_vector_if_needed
-
-                spark = SparkSession.builder.getOrCreate()
-
-                def _get_input_example_as_pd_df():
+                def _get_input_example_df():
                     feature_cols = list(get_feature_cols(input_df, spark_model))
                     limited_input_df = input_df.select(feature_cols).limit(
                         INPUT_EXAMPLE_SAMPLE_ROWS
                     )
-                    return cast_spark_df_with_vector_to_array(limited_input_df).toPandas()
+                    return limited_input_df
 
                 def _infer_model_signature(input_example_slice):
-                    input_slice_df = _find_and_set_features_col_as_vector_if_needed(
-                        spark.createDataFrame(input_example_slice), spark_model
-                    )
-                    model_output = spark_model.transform(input_slice_df).drop(
-                        *input_slice_df.columns
-                    )
+                    prediction_column = _check_or_set_model_prediction_column(spark_model, input_example_slice)
+                    model_output = spark_model.transform(input_example_slice).select(prediction_column)
```

Comment:
Previous code is wrong,

the inferred output signature should keep consistent with `PyFuncModel.predict` outputs.

Reply:
That previous code would have preserved a generated vector column with the output, right?

Reply:
Previous code preserve all new columns generated by `model.transform` and generates output signature according to these new columns.

But this does not match `PyfuncModel.predict` output. `PyfuncModel.predict` only returns one "prediction" column output.

We should make the output signature consistent with `PyfuncModel.predict` output.